app/controllers/websocket_authentication_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class WebsocketAuthenticationController < WebsocketRails::BaseController
+  def authorize_channels
+    if message[:channel].start_with?("job")
+      authorize_job
+    elsif message[:channel].start_with?("worker_info")
+      authorize_worker_info
+    else
+      deny_channel
+    end
+  end
+
+  private
+
+  def authorize_job
+    match = /^job\.(?<id>\w+)$/.match(message[:channel])
+    return deny_channel unless match
+
+    job = Job.find_by(id: match[:id].to_i)
+    if job && current_user && job.user == current_user
+      accept_channel job.as_json
+    else
+      deny_channel
+    end
+  end
+
+  def authorize_worker_info
```

Comment:
Hmm, these two methods `authorize_worker_info`, and `authorize_job` are very similar. Do you think we'd be able to create a helper method to reuse? I'm just thinking about if we want to extend this controller for further channels.

Reply:
Refactored, now it passes the channel prefix and the class to search for into a method